collector/perf_linux.go

```diff
@@ -585,13 +614,10 @@ func (c *perfCollector) Update(ch chan<- prometheus.Metric) error {
 
 func (c *perfCollector) updateHardwareStats(ch chan<- prometheus.Metric) error {
 	for _, profiler := range c.perfHwProfilers {
-		hwProfile, err := (*profiler).Profile()
-		if err != nil {
+		hwProfile := &perf.HardwareProfile{}
```

Comment:
In theory these could come from an object pool so extra allocations aren't required.